src/components/scenes/FioConnectWalletScene.js

```diff
@@ -0,0 +1,171 @@
+// @flow
+
+import type { EdgeCurrencyWallet } from 'edge-core-js'
+import React, { Component } from 'react'
+import { ActivityIndicator, FlatList, Image, ScrollView, TouchableHighlight, View } from 'react-native'
+import { Actions } from 'react-native-router-flux'
+
+import * as Constants from '../../constants/SceneKeys'
+import s from '../../locales/strings.js'
+import { findWalletByFioAddress } from '../../modules/FioAddress/util'
+import T from '../../modules/UI/components/FormattedText/index'
+import { CryptoExchangeWalletListRowStyle as walletStyles } from '../../styles/components/CryptoExchangeWalletListRowStyle'
+import { styles } from '../../styles/scenes/FioConnectWalletStyle'
+import type { FioConnectionWalletItem } from '../../types/types'
+import { SceneWrapper } from '../common/SceneWrapper'
+import { showError } from '../services/AirshipInstance'
+
+export type State = {
+  selectedWallets: { [walletId: string]: FioConnectionWalletItem },
+  fioWallet: EdgeCurrencyWallet | null,
+  fioWalletLoading: boolean
+}
+
+export type FioConnectWalletStateProps = {
+  wallets: {
+    [publicAddressCurrencyCode: string]: FioConnectionWalletItem
+  },
+  fioWallets: EdgeCurrencyWallet[],
+  loading: boolean
+}
+
+export type FioConnectWalletRouteProps = {
+  fioAddressName: string
+}
+
+type Props = FioConnectWalletStateProps & FioConnectWalletRouteProps
+
+export class FioConnectWalletScene extends Component<Props, State> {
+  state = {
+    selectedWallets: {},
+    fioWallet: null,
+    fioWalletLoading: false
+  }
+
+  componentDidMount (): void {
+    this.findFioWallet()
+  }
+
+  findFioWallet = async () => {
+    const { fioAddressName, fioWallets } = this.props
+    this.setState({ fioWalletLoading: true })
+    const fioWallet = await findWalletByFioAddress(fioWallets, fioAddressName)
+    this.setState({ fioWalletLoading: false, fioWallet })
+  }
+
+  _onToggleContinue = async (): Promise<void> => {
+    const { fioAddressName } = this.props
+    const { selectedWallets, fioWallet } = this.state
+    if (fioWallet) {
+      Actions[Constants.FIO_CONNECT_TO_WALLETS_CONFIRM]({ fioAddressName, fioWallet, selectedWallets: Object.values(selectedWallets) })
+    } else {
+      showError(s.strings.fio_connect_wallets_err)
```

Comment:
`fio_wallet_missing_for_fio_address`: `Sys error. Can not find FIO Wallet for FIO Address`

Reply:
changed